model/task/task.go

```diff
@@ -658,25 +660,30 @@ func GenerateNotRun() ([]Task, error) {
 		StartTimeKey:      bson.M{"$gt": time.Now().Add(-maxGenerateTimeAgo)}, // ignore older tasks, just in case
 		GenerateTaskKey:   true,                                               // task contains generate.tasks command
 		GeneratedTasksKey: bson.M{"$exists": false},                           // generate.tasks has not yet run
-		GeneratedJSONKey:  bson.M{"$exists": true},                            // config has been posted by generate.tasks command
+		"$or": []bson.M{
+			bson.M{GeneratedJSONAsStringKey: bson.M{"$exists": true}}, // config has been posted by generate.tasks command
+			bson.M{GeneratedJSONKey: bson.M{"$exists": true}},         // TODO Remove after EVG-6759 is deployed
+		},
 	}))
 }
 
 // SetGeneratedJSON sets JSON data to generate tasks from.
 func (t *Task) SetGeneratedJSON(json []json.RawMessage) error {
-	if len(t.GeneratedJSON) > 0 {
+	if len(t.GeneratedJSONAsString) > 0 || len(t.GeneratedJSON) > 0 {
```

Comment:
If `t.GeneratedJSON` exists but `t.GeneratedJSONAsString` does not, is it worth it to still update GeneratedJSONAsString?

Reply:
This is a sanity check. If either exists, we should not have gotten here, because `*GenerateConnector.GenerateTasks` checks the boolean `t.GeneratedTasks` immediately before calling `*Task.SetGeneratedJSON`. If either `t.GeneratedJSON` or `t.GeneratedJSONAsString` is true, we should not have gotten here, and therefore I think we don't want to set anything.